syntax = "proto3";

package chroma;

import "chromadb/proto/chroma.proto";

message Status {
  string reason = 1;
  int32 code = 2;
}

message CreateCollectionRequest {
  Collection collection = 1;
  optional bool get_or_create = 2;
}

message Collection {
  string id = 1;
  string name = 2;
  optional UpdateMetadata metadata = 3;
}

message CreateCollectionResponse {
  Collection collection = 1;
  Status status = 2;
}

message GetCollectionsRequest {
  optional string id = 1;
  optional string name = 2;
  optional string topic = 3;
}

message GetCollectionsResponse {
  repeated Collection collections = 1;
  Status status = 2;
}

service SysDB {
  rpc CreateCollection(CreateCollectionRequest) returns (CreateCollectionResponse) {}
  rpc GetCollections(GetCollectionsRequest) returns (GetCollectionsResponse) {}
}
